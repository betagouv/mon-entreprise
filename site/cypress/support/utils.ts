export const fr = Cypress.env('language') === 'fr'
export const baseUrl = Cypress.config('baseUrl')
