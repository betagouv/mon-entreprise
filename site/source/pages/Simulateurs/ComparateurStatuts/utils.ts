export type ValueType =
	| string
	| number
	| boolean
	| null
	| Record<string, unknown>
export type BestOption = {
	type: 'sasu' | 'ei' | 'ae'
	value?: ValueType
}
export const getBestOption = (options: BestOption[]) => {
	const sortedOptions = options.sort(
		(option1: BestOption, option2: BestOption) => {
			if (option1.value === null || option1.value === undefined) {
				return 1
			}
			if (option2.value === null || option2.value === undefined) {
				return -1
			}

			if (option1.value === option2.value) return 0
			// console.log(option1.value, option2.value, option1.value > option2.value)

			return option1.value > option2.value ? -1 : 1
		}
	)

	return sortedOptions?.[0]?.type
}
