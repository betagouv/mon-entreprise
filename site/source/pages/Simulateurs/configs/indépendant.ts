import { SimulationConfig } from './types'

export const configIndépendant: SimulationConfig = {
	objectifs: [
		"entreprise . chiffre d'affaires",
		'dirigeant . rémunération . totale',
		'dirigeant . rémunération . net',
		'dirigeant . rémunération . net . après impôt',
	],
	questions: {
		"à l'affiche": [
			{
				label: "Type d'activité",
				dottedName: 'entreprise . activité . nature',
			},
			{
				label: 'Micro-fiscal',
				dottedName: 'entreprise . imposition . régime . micro-entreprise',
			},
			{
				label: 'Date de création',
				dottedName: 'entreprise . date de création',
			},
			{
				label: 'ACRE',
				dottedName: 'dirigeant . exonérations . ACRE',
			},
			{
				label: 'Contrats Madelins',
				dottedName: 'dirigeant . indépendant . cotisations facultatives',
			},
			{
				label: 'Conjoint collaborateur',
				dottedName: 'dirigeant . indépendant . conjoint collaborateur',
			},
			{
				label: 'Impôt sur le revenu',
				dottedName: 'impôt . méthode de calcul',
			},
		],
		'liste noire': [
			'entreprise . charges',
			"entreprise . chiffre d'affaires",
			'entreprise . exercice . début',
			'entreprise . exercice . fin',
			'entreprise . catégorie juridique',
			'entreprise . imposition . régime',
		],
		'non prioritaires': [
			'dirigeant . indépendant . cotisations facultatives',
			'dirigeant . indépendant . IJSS',
			'dirigeant . indépendant . PL . PAMC . IJSS',
			'dirigeant . indépendant . PL . CNAVPL . exonération incapacité',
			'dirigeant . indépendant . cotisations et contributions . exonérations . pension invalidité',
			// 'entreprise . ZFU',
			'entreprise . salariés . effectif',
			'entreprise . activité . débit de tabac',
		],
	},
	'unité par défaut': '€/an',
	situation: {
		'dirigeant . régime social': "'indépendant'",
		'entreprise . imposition': "'IR'",
	},
}

export const configEntrepriseIndividuelle: SimulationConfig = {
	...configIndépendant,
	situation: {
		...configIndépendant.situation,
		'entreprise . imposition': "'IR'",
	},
}

export const configEirl: SimulationConfig = {
	...configIndépendant,
}

export const configEurl: SimulationConfig = {
	...configIndépendant,
	situation: {
		...configIndépendant.situation,
		'entreprise . imposition': "'IS'",
	},
}
