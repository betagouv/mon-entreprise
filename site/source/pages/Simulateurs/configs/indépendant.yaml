objectifs:
  - entreprise . chiffre d'affaires
  - dirigeant . rémunération . totale
  - dirigeant . rémunération . net
  - dirigeant . rémunération . net . après impôt

questions:
  à l'affiche:
    Type d'activité: entreprise . activité
    Micro-fiscal: entreprise . imposition . régime . micro-entreprise
    Date de création: entreprise . date de création
    ACRE: entreprise . exonérations . ACRE
    Contrats Madelins: dirigeant . indépendant . cotisations facultatives
    Conjoint collaborateur: dirigeant . indépendant . conjoint collaborateur
    Impôt sur le revenu: impôt . méthode de calcul
  liste noire:
    - entreprise . charges
    - entreprise . exercice . début
    - entreprise . exercice . fin
    - entreprise . catégorie juridique
    - entreprise . imposition . régime
  non prioritaires:
    - dirigeant . indépendant . cotisations facultatives
    - dirigeant . indépendant . IJSS
    - dirigeant . indépendant . PL . PAMC . IJSS
    - dirigeant . indépendant . PL . CNAVPL . exonération incapacité
    - dirigeant . indépendant . cotisations et contributions . exonérations . pension invalidité
    - entreprise . ZFU
    - entreprise . salariés . effectif
    - entreprise . activité . débit de tabac

unité par défaut: €/an
situation:
  dirigeant . régime social: "'indépendant'"
  entreprise . imposition: "'IR'"
