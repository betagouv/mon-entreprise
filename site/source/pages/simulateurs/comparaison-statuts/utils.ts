import { DottedName } from 'modele-social'
import Engine, { Evaluation } from 'publicodes'

import { StatutType } from '@/components/StatutTag'

export type OptionType = {
	engine: Engine<DottedName>
	name: StatutType
	value: Evaluation
}
export const getBestOption = (options: OptionType[], leastIsBest: boolean) => {
	const sortedOptions = [...options].sort(
		(option1: OptionType, option2: OptionType) => {
			if (option1.value === null || option1.value === undefined) {
				return leastIsBest ? -1 : 1
			}
			if (option2.value === null || option2.value === undefined) {
				return leastIsBest ? 1 : -1
			}

			if (option1.value === option2.value) return 0

			return (option1.value > option2.value ? -1 : 1) * (leastIsBest ? -1 : 1)
		}
	)

	return sortedOptions?.[0]?.name
}
