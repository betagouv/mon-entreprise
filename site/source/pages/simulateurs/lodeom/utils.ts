import { sumAll } from 'effect/Number'
import { DottedName } from 'modele-social'
import Engine from 'publicodes'

import { Situation } from '@/domaine/Situation'

// TODO: remplacer "salarié . cotisations . assiette" par "salarié . rémunération . brut"
// lorsqu'elle n'incluera plus les frais professionnels.
export const rémunérationBruteDottedName = 'salarié . cotisations . assiette'
export const lodeomDottedName =
	'salarié . cotisations . exonérations . lodeom . montant'
export const heuresSupplémentairesDottedName =
	'salarié . temps de travail . heures supplémentaires'
export const heuresComplémentairesDottedName =
	'salarié . temps de travail . heures complémentaires'

export type Répartition = {
	IRC: number
	Urssaf: number
}

export type MonthState = {
	rémunérationBrute: number
	options: Options
	lodeom: {
		value: number
		répartition: Répartition
	}
	régularisation: {
		value: number
		répartition: Répartition
	}
}

export type Options = {
	heuresSupplémentaires: number
	heuresComplémentaires: number
	rémunérationETP: number
	rémunérationPrimes: number
	rémunérationHeuresSup: number
}

export type RégularisationMethod = 'annuelle' | 'progressive'

const getDateForContexte = (monthIndex: number, year: number): string => {
	const date = new Date(year, monthIndex)

	return date.toLocaleDateString('fr')
}

const getMonthlyLodeom = (
	date: string,
	rémunérationBrute: number,
	options: Options,
	engine: Engine<DottedName>
): number => {
	const lodeom = engine.evaluate({
		valeur: lodeomDottedName,
		unité: '€/mois',
		contexte: {
			date,
			[rémunérationBruteDottedName]: rémunérationBrute,
			[heuresSupplémentairesDottedName]: options.heuresSupplémentaires,
			[heuresComplémentairesDottedName]: options.heuresComplémentaires,
		},
	})

	return lodeom.nodeValue as number
}

const getTotalLodeom = (
	rémunérationBrute: number,
	SMIC: number,
	coefT: number,
	engine: Engine<DottedName>
): number => {
	const lodeom = engine.evaluate({
		valeur: lodeomDottedName,
		arrondi: 'non',
		contexte: {
			[rémunérationBruteDottedName]: rémunérationBrute,
			'salarié . temps de travail . SMIC': SMIC,
			'salarié . cotisations . exonérations . T': coefT,
		},
	})

	return lodeom.nodeValue as number
}

const emptyRépartition = {
	IRC: 0,
	Urssaf: 0,
}

const getRépartition = (
	rémunération: number,
	réduction: number,
	engine: Engine<DottedName>
): Répartition => {
	const contexte = {
		[rémunérationBruteDottedName]: rémunération,
		[lodeomDottedName]: réduction,
	}
	const IRC =
		(engine.evaluate({
			valeur: `${lodeomDottedName} . imputation retraite complémentaire`,
			unité: '€/mois',
			contexte,
		})?.nodeValue as number) ?? 0
	const Urssaf =
		(engine.evaluate({
			valeur: `${lodeomDottedName} . imputation sécurité sociale`,
			unité: '€/mois',
			contexte,
		})?.nodeValue as number) ?? 0

	return {
		IRC,
		Urssaf,
	}
}

export const getInitialLodeomMoisParMois = (
	year: number,
	engine: Engine<DottedName>
): MonthState[] => {
	const rémunérationBrute =
		(engine.evaluate({
			valeur: rémunérationBruteDottedName,
			arrondi: 'oui',
			unité: '€/mois',
		})?.nodeValue as number) ?? 0
	const heuresSupplémentaires =
		(engine.evaluate({
			valeur: heuresSupplémentairesDottedName,
			unité: 'heures/mois',
		})?.nodeValue as number) ?? 0
	const heuresComplémentaires =
		(engine.evaluate({
			valeur: heuresComplémentairesDottedName,
			unité: 'heures/mois',
		})?.nodeValue as number) ?? 0
	const rémunérationETP = 0
	const rémunérationPrimes = 0
	const rémunérationHeuresSup = 0

	if (!rémunérationBrute) {
		return Array(12).fill({
			rémunérationBrute,
			options: {
				heuresSupplémentaires,
				heuresComplémentaires,
				rémunérationETP,
				rémunérationPrimes,
				rémunérationHeuresSup,
			},
			lodeom: {
				value: 0,
				répartition: emptyRépartition,
			},
			régularisation: {
				value: 0,
				répartition: emptyRépartition,
			},
		}) as MonthState[]
	}

	return Array.from({ length: 12 }, (_item, monthIndex) => {
		const date = getDateForContexte(monthIndex, year)

		const lodeom = getMonthlyLodeom(
			date,
			rémunérationBrute,
			{
				heuresSupplémentaires,
				heuresComplémentaires,
				rémunérationETP,
				rémunérationPrimes,
				rémunérationHeuresSup,
			},
			engine
		)
		const répartition = getRépartition(rémunérationBrute, lodeom, engine)

		return {
			rémunérationBrute,
			options: {
				heuresSupplémentaires,
				heuresComplémentaires,
				rémunérationETP,
				rémunérationPrimes,
				rémunérationHeuresSup,
			},
			lodeom: {
				value: lodeom,
				répartition,
			},
			régularisation: {
				value: 0,
				répartition: emptyRépartition,
			},
		}
	})
}

export const reevaluateLodeomMoisParMois = (
	data: MonthState[],
	engine: Engine<DottedName>,
	year: number,
	régularisationMethod: RégularisationMethod
): MonthState[] => {
	const totalRémunérationBrute = sumAll(
		data.map((monthData) => monthData.rémunérationBrute)
	)

	if (!totalRémunérationBrute) {
		return data.map((monthData) => {
			return {
				...monthData,
				lodeom: {
					value: 0,
					répartition: emptyRépartition,
				},
				régularisation: {
					value: 0,
					répartition: emptyRépartition,
				},
			}
		})
	}

	const rémunérationBruteCumulées = getRémunérationBruteCumulées(data)
	const SMICCumulés = getSMICCumulés(data, year, engine)
	// Si on laisse l'engine calculer T dans le calcul de l'exonération Lodeom,
	// le résultat ne sera pas bon à cause de l'assiette de cotisations du contexte
	const coefT = engine.evaluate({
		valeur: 'salarié . cotisations . exonérations . T',
	}).nodeValue as number

	const reevaluatedData = data.reduce(
		(reevaluatedData: MonthState[], monthState, monthIndex) => {
			const rémunérationBrute = monthState.rémunérationBrute
			const options = monthState.options
			const lodeom = {
				value: 0,
				répartition: emptyRépartition,
			}
			const régularisation = {
				value: 0,
				répartition: emptyRépartition,
			}

			if (!rémunérationBrute) {
				return [
					...reevaluatedData,
					{
						rémunérationBrute,
						options,
						lodeom,
						régularisation,
					},
				]
			}

			if (régularisationMethod === 'progressive') {
				// La régularisation progressive du mois N est la différence entre l'exonération Lodeom
				// calculée pour la rémunération totale jusqu'à N (comparée au SMIC équivalent pour ces N mois)
				// et la somme des N-1 exonérations déjà accordées (en incluant les régularisations).
				const lodeomTotale = getTotalLodeom(
					rémunérationBruteCumulées[monthIndex],
					SMICCumulés[monthIndex],
					coefT,
					engine
				)
				const lodeomCumulée = sumAll(
					reevaluatedData.map(
						(monthData) =>
							monthData.lodeom.value + monthData.régularisation.value
					)
				)
				régularisation.value = lodeomTotale - lodeomCumulée

				if (régularisation.value > 0) {
					lodeom.value = régularisation.value
					lodeom.répartition = getRépartition(
						rémunérationBrute,
						lodeom.value,
						engine
					)
					régularisation.value = 0
				} else if (régularisation.value < 0) {
					régularisation.répartition = getRépartition(
						rémunérationBrute,
						régularisation.value,
						engine
					)
				}
			} else if (régularisationMethod === 'annuelle') {
				const date = getDateForContexte(monthIndex, year)
				lodeom.value = getMonthlyLodeom(
					date,
					rémunérationBrute,
					options,
					engine
				)

				if (monthIndex === data.length - 1) {
					// La régularisation annuelle est la différence entre l'exonération Lodeom calculée
					// pour la rémunération annuelle (comparée au SMIC annuel) et la somme des exonérations
					// Lodeom déjà accordées.
					const lodeomTotale = getTotalLodeom(
						rémunérationBruteCumulées[monthIndex],
						SMICCumulés[monthIndex],
						coefT,
						engine
					)
					const currentLodeomCumulée =
						lodeom.value +
						sumAll(reevaluatedData.map((monthData) => monthData.lodeom.value))
					régularisation.value = lodeomTotale - currentLodeomCumulée

					if (lodeom.value + régularisation.value > 0) {
						lodeom.value += régularisation.value
						lodeom.répartition = getRépartition(
							rémunérationBrute,
							lodeom.value,
							engine
						)
						régularisation.value = 0
					} else if (régularisation.value < 0) {
						régularisation.répartition = getRépartition(
							rémunérationBrute,
							régularisation.value,
							engine
						)
					}
				}
			}

			return [
				...reevaluatedData,
				{
					rémunérationBrute,
					options,
					lodeom,
					régularisation,
				},
			]
		},
		[]
	)

	return reevaluatedData
}

const getSMICCumulés = (
	data: MonthState[],
	year: number,
	engine: Engine<DottedName>
): number[] => {
	return data.reduce((SMICCumulés: number[], monthData, monthIndex) => {
		// S'il n'y a pas de rémunération ce mois-ci, il n'y a pas d'exonération Lodeom
		// et il ne faut pas compter le SMIC de ce mois-ci dans le SMIC cumulé.
		if (!monthData.rémunérationBrute) {
			SMICCumulés.push(0)

			return SMICCumulés
		}

		const contexte = {
			date: getDateForContexte(monthIndex, year),
		} as Situation

		if (!monthData.options.rémunérationETP) {
			contexte[heuresSupplémentairesDottedName] =
				monthData.options.heuresSupplémentaires
			contexte[heuresComplémentairesDottedName] =
				monthData.options.heuresComplémentaires
		}

		let SMIC = engine.evaluate({
			valeur: 'salarié . temps de travail . SMIC',
			unité: '€/mois',
			contexte,
		}).nodeValue as number

		if (monthData.options.rémunérationETP) {
			const SMICHoraire = engine.evaluate({
				valeur: 'SMIC . horaire',
				contexte,
			}).nodeValue as number
			// On retranche les primes et le paiements des heures supplémentaires à la rémunération versée
			// et on la compare à la rémunération équivalente "mois complet" sans les primes
			const prorata =
				(monthData.rémunérationBrute -
					monthData.options.rémunérationPrimes -
					monthData.options.rémunérationHeuresSup) /
				monthData.options.rémunérationETP
			// On applique ce prorata au SMIC mensuel et on y ajoute les heures supplémentaires et complémentaires
			SMIC =
				SMIC * prorata +
				SMICHoraire *
					(monthData.options.heuresSupplémentaires +
						monthData.options.heuresComplémentaires)
		}

		let SMICCumulé = SMIC
		if (monthIndex > 0) {
			// Il faut aller chercher la dernière valeur positive du SMIC cumulé.
			const previousSMICCumulé =
				SMICCumulés.findLast((SMICCumulé) => SMICCumulé > 0) ?? 0
			SMICCumulé = previousSMICCumulé + SMIC
		}

		SMICCumulés.push(SMICCumulé)

		return SMICCumulés
	}, [])
}

const getRémunérationBruteCumulées = (data: MonthState[]) => {
	return data.reduce(
		(rémunérationBruteCumulées: number[], monthData, monthIndex) => {
			// S'il n'y a pas de rémunération ce mois-ci, il n'y a pas d'exonération Lodeom
			// et elle ne compte pas non plus pour la régularisation des mois à venir.
			if (!monthData.rémunérationBrute) {
				rémunérationBruteCumulées.push(0)

				return rémunérationBruteCumulées
			}

			let rémunérationBruteCumulée = monthData.rémunérationBrute
			if (monthIndex > 0) {
				// Il faut aller chercher la dernière valeur positive de la rémunération cumulée.
				const previousRémunérationBruteCumulée =
					rémunérationBruteCumulées.findLast(
						(rémunérationBruteCumulée) => rémunérationBruteCumulée > 0
					) ?? 0
				rémunérationBruteCumulée =
					previousRémunérationBruteCumulée + monthData.rémunérationBrute
			}

			rémunérationBruteCumulées.push(rémunérationBruteCumulée)

			return rémunérationBruteCumulées
		},
		[]
	)
}
