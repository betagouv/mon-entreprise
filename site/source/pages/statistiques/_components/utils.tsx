export function formatDay(date: string | Date, language: string) {
	return new Date(date).toLocaleString(language, {
		weekday: 'long',
		day: 'numeric',
		month: 'long',
	})
}
export function formatMonth(date: string | Date, language: string) {
	return new Date(date).toLocaleString(language, {
		month: 'long',
		year: 'numeric',
	})
}
export function formatIndicator(number: number, language: string) {
	return Intl.NumberFormat(language, {
		notation: 'compact',
	}).format(number)
}

export function formatProgression(
	previousValue: number,
	currentValue: number,
	language: string
) {
	return Intl.NumberFormat(language, {
		style: 'percent',
		signDisplay: 'exceptZero',
	}).format(currentValue / previousValue - 1)
}
export function emojiSatisfaction(satisfaction: number) {
	return satisfaction > 0.85
		? '😀'
		: satisfaction > 0.65
		? '🙂'
		: satisfaction > 0.5
		? '😐'
		: '😕'
}

export function messageTypeSatisfaction(satisfaction: number) {
	return satisfaction > 0.85
		? 'success'
		: satisfaction > 0.65
		? 'primary'
		: satisfaction > 0.5
		? 'info'
		: 'error'
}
