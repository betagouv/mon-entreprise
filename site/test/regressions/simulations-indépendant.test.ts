import { configIndépendant } from '@/pages/Simulateurs/configs/indépendant'
import { DottedName } from 'modele-social'
import { it } from 'vitest'
import independentSituations from './simulations-indépendant.yaml'
import { runSimulations } from './utils'

it('calculate simulations-indépendant', () => {
	const objectifs = [
		'dirigeant . rémunération . totale',
		'dirigeant . rémunération . cotisations',
		'dirigeant . rémunération . net',
		'dirigeant . indépendant . revenu professionnel',
		'impôt . montant',
		'dirigeant . rémunération . net . après impôt',
		'entreprise . charges',
		"entreprise . chiffre d'affaires",
		'dirigeant . indépendant . cotisations et contributions . début activité',
	] as DottedName[]
	runSimulations(independentSituations, objectifs, configIndépendant.situation)
})
