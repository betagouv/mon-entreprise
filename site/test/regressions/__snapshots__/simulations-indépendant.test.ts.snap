// Vitest Snapshot v1

exports[`calculate simulations-indépendant > acre 1`] = `"[73019,23019,50000,51980,9516,40484,0,73019,1408]"`;

exports[`calculate simulations-indépendant > activité 1`] = `"[29086,9086,20000,20787,898,19102,0,29086,0]"`;

exports[`calculate simulations-indépendant > activité 2`] = `"[29102,9102,20000,20787,898,19102,0,29102,0]"`;

exports[`calculate simulations-indépendant > conjoint collaborateur 1`] = `"[77175,27175,50000,52092,9550,40450,0,77175,0]"`;

exports[`calculate simulations-indépendant > conjoint collaborateur 2`] = `"[78236,28236,50000,52121,9558,40442,0,78236,0]"`;

exports[`calculate simulations-indépendant > conjoint collaborateur 3`] = `"[80757,30757,50000,52189,9579,40421,0,80757,0]"`;

exports[`calculate simulations-indépendant > conjoint collaborateur 4`] = `"[75298,25298,50000,52041,9534,40466,0,75298,0]"`;

exports[`calculate simulations-indépendant > conjoint collaborateur 5`] = `"[75297,25297,50000,52041,9534,40466,0,75297,0]"`;

exports[`calculate simulations-indépendant > conjoint collaborateur 6`] = `
"[652221,152221,500000,517706,220485,279515,0,652221,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;

exports[`calculate simulations-indépendant > contrats madelin 1`] = `"[30000,13181,16819,17881,433,16386,10000,40000,0]"`;

exports[`calculate simulations-indépendant > contrats madelin 2`] = `"[30000,14381,15619,17881,433,15186,10000,40000,0]"`;

exports[`calculate simulations-indépendant > contrats madelin 3`] = `"[30000,10381,19619,20431,841,18778,10000,40000,0]"`;

exports[`calculate simulations-indépendant > contrats madelin 4`] = `"[20000,6732,13268,13809,0,13268,1000,21000,0]"`;

exports[`calculate simulations-indépendant > contrats madelin 5`] = `
"[300000,79617,220383,228526,82145,138238,15000,315000,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;

exports[`calculate simulations-indépendant > contrats madelin 6`] = `"[20000,6832,13168,13709,0,13168,500,20500,0]"`;

exports[`calculate simulations-indépendant > contrats madelin 7`] = `"[20000,13132,6868,9695,0,6868,0,20000,0]"`;

exports[`calculate simulations-indépendant > cotisations facultatives 1`] = `"[30000,13181,16819,17881,433,16386,0,30000,0]"`;

exports[`calculate simulations-indépendant > cotisations facultatives 2`] = `"[30000,14381,15619,17881,433,15186,0,30000,0]"`;

exports[`calculate simulations-indépendant > cotisations facultatives 3`] = `"[30000,10381,19619,20431,841,18778,0,30000,0]"`;

exports[`calculate simulations-indépendant > cotisations facultatives 4`] = `"[20000,6732,13268,13809,0,13268,0,20000,0]"`;

exports[`calculate simulations-indépendant > cotisations facultatives 5`] = `
"[300000,79617,220383,228526,82145,138238,0,300000,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;

exports[`calculate simulations-indépendant > cotisations facultatives 6`] = `
"[300000,83117,216883,225026,80570,136313,0,300000,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;

exports[`calculate simulations-indépendant > cotisations minimales 1`] = `"[1375,1275,100,135,0,100,0,1375,0]"`;

exports[`calculate simulations-indépendant > cotisations minimales 2`] = `"[244,144,100,104,0,100,0,244,0]"`;

exports[`calculate simulations-indépendant > exonération covid 1`] = `"[71347,21347,50000,51815,9550,40450,0,71347,2059]"`;

exports[`calculate simulations-indépendant > exonération covid 2`] = `"[500,12977,50000,51032,9315,40685,0,62977,103]"`;

exports[`calculate simulations-indépendant > exonération covid 3`] = `"[50000,17475,50000,51440,9438,40562,0,67475,141]"`;

exports[`calculate simulations-indépendant > exonération pension invalidité 1`] = `
"[45000,10908,34092,35311,4515,29577,0,45000,0]
Notifications affichées : dirigeant . indépendant . cotisations et contributions . exonérations . pension invalidité"
`;

exports[`calculate simulations-indépendant > imposition à l'IS 1`] = `
"[100000,30100,69900,72612,13527,56373,0,100000,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;

exports[`calculate simulations-indépendant > imposition à l'IS 2`] = `
"[100000,30100,69900,72612,13527,56373,0,100000,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;

exports[`calculate simulations-indépendant > impôt sur le revenu 1`] = `"[29086,9086,20000,20787,603,19397,0,29086,0]"`;

exports[`calculate simulations-indépendant > impôt sur le revenu 2`] = `"[73019,23019,50000,51980,8213,41787,0,73019,0]"`;

exports[`calculate simulations-indépendant > impôt sur le revenu 3`] = `"[29086,9086,20000,20787,2079,17921,0,29086,0]"`;

exports[`calculate simulations-indépendant > inversions 1`] = `"[2000,1386,614,666,0,614,0,2000,0]"`;

exports[`calculate simulations-indépendant > inversions 2`] = `"[50000,16004,33996,35351,4527,29469,0,50000,0]"`;

exports[`calculate simulations-indépendant > inversions 3`] = `"[14597,4597,10000,10394,0,10000,0,14597,0]"`;

exports[`calculate simulations-indépendant > inversions 4`] = `"[72000,22703,49297,51249,9297,40000,0,72000,0]"`;

exports[`calculate simulations-indépendant > inversions 5`] = `"[14597,4597,10000,10394,0,10000,1000,15597,0]"`;

exports[`calculate simulations-indépendant > inversions 6`] = `"[19000,5929,13071,13584,0,13071,1000,20000,0]"`;

exports[`calculate simulations-indépendant > inversions 7`] = `"[18000,5627,12373,12860,0,12373,2000,20000,0]"`;

exports[`calculate simulations-indépendant > échelle de revenus 1`] = `"[1861,1361,500,548,0,500,0,1861,0]"`;

exports[`calculate simulations-indépendant > échelle de revenus 2`] = `"[2466,1466,1000,1064,0,1000,0,2466,0]"`;

exports[`calculate simulations-indépendant > échelle de revenus 3`] = `"[3075,1575,1500,1581,0,1500,0,3075,0]"`;

exports[`calculate simulations-indépendant > échelle de revenus 4`] = `"[3683,1683,2000,2097,0,2000,0,3683,0]"`;

exports[`calculate simulations-indépendant > échelle de revenus 5`] = `"[7428,2428,5000,5199,0,5000,0,7428,0]"`;

exports[`calculate simulations-indépendant > échelle de revenus 6`] = `"[14597,4597,10000,10394,0,10000,0,14597,0]"`;

exports[`calculate simulations-indépendant > échelle de revenus 7`] = `
"[139589,39589,100000,103788,28275,71725,0,139589,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;

exports[`calculate simulations-indépendant > échelle de revenus 8`] = `
"[1239948,239948,1000000,1033666,473305,526695,0,1239948,0]
Notifications affichées : entreprise . TVA . franchise de TVA . seuils dépassés"
`;
